src/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi

```diff
@@ -233,18 +234,31 @@ cdef void _call(
   cdef grpc_call_error c_call_error
   cdef tuple error_and_wrapper_tag
   cdef _BatchOperationTag wrapper_tag
+  cdef void *registered_call_handle
+  cdef const char *host_ptr
   with channel_state.condition:
     if channel_state.open:
       method_slice = _slice_from_bytes(method)
       if host is None:
         host_slice_ptr = NULL
+        host_ptr = NULL
       else:
         host_slice = _slice_from_bytes(host)
         host_slice_ptr = &host_slice
-      call_state.c_call = grpc_channel_create_call(
-          channel_state.c_channel, NULL, flags,
-          c_completion_queue, method_slice, host_slice_ptr,
-          _timespec_from_time(deadline), NULL)
+        host_ptr = <const char *>host
+      if registered_method:
+        registered_call_handle = grpc_channel_register_call(
```

Comment:
We'll want to get clarification from @ctiller on this ([reference](https://github.com/grpc/grpc/pull/1261)), but I don't think the intent with this function is to call it on a per-call basis, but instead, to call it on a per *channel* basis, with the idea being that it saves on per-call string operations on the method/service.

However, I know that the primary value we intend to get from this is registration for the purposes of observability, not a cost-savings on string operations. And it would take some work so that we plumb through a pre-registration handle rather than a bool. So it might be okay to leave things as they are, but something to think about.

If you did decide to do this restructure, I think it would look like this:

- Add a new gRPC Python channel method called something like `_create_registered_method` that returns an object that's basically a Python int containing the `void*` returned by a call to `grpc_channel_register_call`
- Call `_created_registered_method` once for each method in the `Stub` class constructor and pass the returned Python integer through to  `unary_unary`, `unary_stream`, etc.
- Plumb the handle down here to `_call` if supplied
-  if the handle is supplied, call `grpc_channel_create_registered_call` with it. If not, call `grpc_channel_create_call`.

If you don't do the restructure, you _should_ make sure that repeatedly calling `grpc_channel_register_call` many times repeatedly is not going to cause issues for us, for example unbounded memory growth.